### PR TITLE
Add preview of images in Apps catalog

### PR DESCRIPTION
**Pull request by makeevrserg**
**Background**

Currently int application users doesn't have an ability to preiew screenshots of hub applications. 

**Changes**

- Created module screenshotspreview which allows to display preview images
- Added module integration with modules inside faphub via callbacks

**Test plan**

1. Open Hub->Apps
2. Wait until apps loaded and click on the image inside app item
3. Open app screen details and click on the image
4. Open Hub->Apps->Search and repeat the steps 2-3 from above

### CODE REVIEW

File: components/bottombar/api/src/main/java/com/flipperdevices/bottombar/api/BottomBarDecomposeComponent.kt [hunk 1]
```diff
@@ -12,7 +13,8 @@ abstract class BottomBarDecomposeComponent<C : Any> : CompositeDecomposeComponen
         operator fun invoke(
             componentContext: ComponentContext,
             onBack: DecomposeOnBackParameter,
-            deeplink: Deeplink.BottomBar?
+            deeplink: Deeplink.BottomBar?,
+            screenshotsClickListener: ScreenshotsClickListener
```
Comment (makeevrserg) [inline components/bottombar/api/src/main/java/com/flipperdevices/bottombar/api/BottomBarDecomposeComponent.kt:17]: There's overall "little bit" callback hell right now. But even deeplinks will be required inside AssistedFactories

Comment (makeevrserg) [inline components/bottombar/api/src/main/java/com/flipperdevices/bottombar/api/BottomBarDecomposeComponent.kt:17]: There's overall "little bit" callback hell right now. But even deeplinks will be required inside AssistedFactories

Comment (LionZXY) [reply components/bottombar/api/src/main/java/com/flipperdevices/bottombar/api/BottomBarDecomposeComponent.kt:17]: I think you should use `LocalRootNavigation` in this case. Could you, please, check this?

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/ComposableFullScreenshotScreen.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+@Composable
+internal fun ComposableFullScreenshotScreen(
+    onBack: () -> Unit,
+    title: String,
+    selected: Int,
+    screenshots: ImmutableList<String>,
+    urlImageShareViewModel: UrlImageShareViewModel
+) {
+    val itemsAmount = remember { screenshots.size }
```
Comment (LionZXY) [inline components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/ComposableFullScreenshotScreen.kt:30]: Are you sure that current keys in `remember` is correct? Maybe use `screenshots` as key?
Also i'm not sure if it's really necessary to use remember here. Can you check?

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/content/ComposableFullScreenshotAppBar.kt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            .fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Icon(
+            modifier = Modifier
+                .size(24.dp)
+                .clickableRipple(bounded = false, onClick = onBack),
+            painter = painterResource(DesignSystem.drawable.ic_back),
+            contentDescription = null.orEmpty()
```
Comment (LionZXY) [inline components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/content/ComposableFullScreenshotAppBar.kt:52]: Why not use `null` here?

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/viewmodel/UrlImageShareViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+) : DecomposeViewModel() {
+
+    private fun shareScreenshot(bitmap: Bitmap) {
+        val date = SimpleDateFormat(TIMEFORMAT, Locale.US).format(Date())
+        val filename = "$SCREENSHOT_FILE_PREFIX-$date.png"
+        val sharableFile = SharableFile(applicationContext, filename)
+        sharableFile.createClearNewFileWithMkDirs()
+        sharableFile.outputStream().use { fis ->
+            bitmap.compress(Bitmap.CompressFormat.PNG, QUALITY, fis)
+        }
```
Comment (LionZXY) [inline components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/viewmodel/UrlImageShareViewModel.kt:41]: Are you sure you want to copy and encode a file in a UI thread?

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/viewmodel/UrlImageShareViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        url.decodeBitmap()
+            .map { bitmap -> bitmap.fillBackground() }
+            .onSuccess { bitmap -> withContext(Dispatchers.Main) { shareScreenshot(bitmap) } }
+    }
+
+    // Images have transparent background thus we need to fill it with accent color
+    private fun Bitmap.fillBackground(): Bitmap {
+        val newBitmap = Bitmap.createBitmap(width, height, config)
+        val canvas = Canvas(newBitmap)
+        canvas.drawColor(Color.rgb(ACCENT_COLOR_R, ACCENT_COLOR_G, ACCENT_COLOR_B))
```
Comment (LionZXY) [inline components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/viewmodel/UrlImageShareViewModel.kt:65]: Maybe overuse the colour from other place?

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/content/ComposableFullScreenshotAppBar.kt (hunk 3 above)
Comment (makeevrserg) [reply components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/content/ComposableFullScreenshotAppBar.kt]: Added contentDescription

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/ComposableFullScreenshotScreen.kt (hunk 2 above)
Comment (makeevrserg) [reply components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/composable/ComposableFullScreenshotScreen.kt]: Previously there was different chunk of code. Indeed in ImmutableList the `size` is set at init and can't be changed after. So it's useless now

File: components/bottombar/api/src/main/java/com/flipperdevices/bottombar/api/BottomBarDecomposeComponent.kt (hunk 1 above)
Comment (makeevrserg) [reply components/bottombar/api/src/main/java/com/flipperdevices/bottombar/api/BottomBarDecomposeComponent.kt]: Replaced with `LocalRootNavigation`

File: components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/viewmodel/UrlImageShareViewModel.kt (hunk 5 above)
Comment (makeevrserg) [reply components/faphub/screenshotspreview/impl/src/main/kotlin/com/flipperdevices/faphub/screenshotspreview/impl/viewmodel/UrlImageShareViewModel.kt]: Replaced with accent color from DesignSystem

**Review by LionZXY — APPROVED** [review]
Thank you so much for your work - it's of incredible quality both in terms of code and final user experience!

**Pull request merged**